youtube_dl/extractor/abc.py

```diff
@@ -103,6 +103,65 @@ def _real_extract(self, url):
         }
 
 
+class ABCIViewShowIE(InfoExtractor):
+    IE_NAME = 'abc.net.au:iview:show'
+    _VALID_URL = r'https?://iview\.abc\.net\.au/(?:[^/]+/)*show/(?P<id>[^/?#]+)/?'
```

Comment:
Remove useless `/?` from the end.

Reply:
Not useless; distinguishes shows from episodes which use a different extractor.

Reply:
_VALID_URL is used by re.match() - this checks for a match only at the beginning of the string. You can remove the `/?` and the pattern will still match correctly.

See: https://docs.python.org/3/library/re.html#search-vs-match

Reply:
Removing the `/?` causes the tests to fail because the sample URL used in `ABCIViewIE` (different class in the same file) also matches this URL. Need to avoid matching `/show/:name/series/0/video/:id`.